flink/v1.16/flink/src/main/java/org/apache/iceberg/flink/FlinkWriteConf.java

```diff
@@ -153,16 +167,89 @@ public String orcCompressionStrategy() {
         .parse();
   }
 
-  public DistributionMode distributionMode() {
```

Comment:
I understand the motivation. but it makes the `FlinkWriteConf` not a simple config hierarchy resolution class. it puts more complex logic in the conf class. hence I am not sure if this is better or not.

Reply:
On reflection, I removed this logic and kept it in FlinkSink to keep the `FlinkWriteConf` pure.